spec/eth/client_spec.rb

```diff
@@ -424,4 +438,76 @@
       expect(geth_ipc.is_valid_signature(contract, hashed, signature)).to be true
     end
   end
+
+  describe ".send" do
+    it "should set up the WebSocket connection" do
+      expect(geth_dev_ws.instance_variable_get("@ws")).to be_instance_of(WebSocket::Client::Simple::Client)
+    end
+
+    it "should send a message to the WebSocket server and receive a response" do
+      payload = {
+        id: 1,
+        jsonrpc: "2.0",
+        method: "eth_subscribe",
+        params: ["newHeads"],
+      }
+      received_data = nil
+
+      geth_dev_ws.instance_variable_get("@ws").on :message do |msg|
+        received_data = JSON.parse(msg.data)
+      end
+
+      # Wait for the connection to be established
+      start_time = Time.now
+      loop do
+        break if geth_dev_ws.instance_variable_get("@ws").open? || (Time.now - start_time > 3)
+      end
+
+      geth_dev_ws.send_request(payload)
+
+      # Wait for the response to be received
+      start_time = Time.now
+      loop do
+        break if received_data || (Time.now - start_time > 3)
```

Comment:
It might be good to add a TIMEOUT to the client, so we don't have to deal with this externally